diff --git a/model/model-flow-base/src/main/yang/opendaylight-action-types.yang b/model/model-flow-base/src/main/yang/opendaylight-action-types.yang
index 6f2aa63..ea936ed 100644
--- a/model/model-flow-base/src/main/yang/opendaylight-action-types.yang
+++ b/model/model-flow-base/src/main/yang/opendaylight-action-types.yang
@@ -59,6 +59,15 @@ module opendaylight-action-types {
                     leaf max-length {
                         type uint16;
                     }
+                    leaf wavelength {
+                        type uint16;
+                    }
+
+                    leaf timeslot {
+                        type string {
+                            pattern '[01]{80}';
+                        }
+                    }
                 }
             }
 
diff --git a/model/model-flow-base/src/main/yang/opendaylight-match-types.yang b/model/model-flow-base/src/main/yang/opendaylight-match-types.yang
index 332f27e..42cc609 100644
--- a/model/model-flow-base/src/main/yang/opendaylight-match-types.yang
+++ b/model/model-flow-base/src/main/yang/opendaylight-match-types.yang
@@ -296,6 +296,16 @@ module opendaylight-match-types {
             type inv:node-connector-id;
         }
 
+        leaf wavelength {
+            type uint16;
+        }
+
+        leaf timeslot {
+            type string {
+                pattern '[01]{80}';
+            }
+        }
+
         container "metadata" {
             uses of-metadata;
         }
@@ -369,4 +379,4 @@ module opendaylight-match-types {
             uses "tcp-flag-match-fields";
         }
     }
-}
\ No newline at end of file
+}
